lib/rules/sort-alternatives.ts

```diff
@@ -157,25 +158,44 @@ function approximateLexicographicallySmallest(
     return getLexicographicallySmallestFromCharSets(prefix)
 }
 
+function getLexicographicallySmallestFromAlternative(
+    alternative: Alternative,
+    parser: JS.Parser,
+    flags: ReadonlyFlags,
+): Word | undefined
+function getLexicographicallySmallestFromAlternative(
+    alternative: StringAlternative,
+    parser: JS.Parser,
+    flags: ReadonlyFlags,
+): Word
 /**
  * If defined, this will return the lexicographically smallest string accepted
  * by the given alternative (ignoring assertions).
  */
 function getLexicographicallySmallestFromAlternative(
-    alternative: Alternative,
+    alternative: Alternative | StringAlternative,
     parser: JS.Parser,
     flags: ReadonlyFlags,
 ): Word | undefined {
-    const { elements } = alternative
-    if (isOnlyCharacters(elements)) {
+    if (
+        alternative.type === "StringAlternative" ||
+        hasOnlyCharacters(alternative)
+    ) {
+        const elements = alternative.elements
         // fast path to avoid converting simple alternatives into NFAs
         const smallest: Word = []
         for (const e of elements) {
-            // FIXME: TS Error
-            // @ts-expect-error -- FIXME
-            const cs = toCharSet(e, flags)
-            if (cs.isEmpty) return undefined
-            smallest.push(cs.ranges[0].min)
+            const cs = toUnicodeSet(e, flags)
+            const firstCharSets = [
+                ...(cs.chars.isEmpty ? [] : [cs.chars]),
+                ...cs.accept.wordSets
+                    .map((ws) => (ws.length ? ws[0] : null))
+                    .filter((wcs): wcs is CharSet => Boolean(wcs)),
+            ]
+            if (!firstCharSets.length) return undefined
+            smallest.push(
+                Math.min(...firstCharSets.map((fcs) => fcs.ranges[0].min)),
+            )
```

Comment:
This isn't correct. Due to `ws[0]`, this function will incorrectly say that the lexicographically smallest word of `a[\q{bc}]d` is `abd`.

Doing this correctly isn't easy, because we have to account for the next characters in the alternative. We can't just pick the lexicographically smallest word in the current Unicode set. E.g. the lexicographically smallest word in `[\q{a|aba}]` is `a`, but the lexicographically smallest word `[\q{a|aba}]bb` is `ababb`.  So we have to construct the smallest word by going through the elements back to front.

Of course, if the current Unicode set doesn't contain strings, then we don't have to account for the characters after it, so we can optimize the common case.

Reply:
Thank you for checking this PR!
I tried to fix the problem. Could you please check again?